test/apps/iodemo/io_demo.cc

```diff
@@ -251,47 +348,85 @@ class P2pDemoCommon : public UcxContext {
         send_recv_data_as_chunks(conn, data_size, sn, XFER_TYPE_RECV, callback);
     }
 
-    uint32_t get_chunk_cnt(size_t data_size) {
-        return (data_size + opts().chunk_size - 1) / opts().chunk_size;
+    static uint32_t get_chunk_cnt(size_t data_size, size_t chunk_size) {
+        return (data_size + chunk_size - 1) / chunk_size;
     }
-    
+
     void send_data(UcxConnection* conn, size_t data_size, uint32_t sn,
                    UcxCallback* callback = EmptyCallback::get()) {
+        if (opts().validate) {
+            chk_hdr_t *dhdr  = reinterpret_cast<chk_hdr_t*>(buffer());
+            tr_hdr_t *thdr   = reinterpret_cast<tr_hdr_t*>(dhdr + 1);
+            uint8_t *payload = reinterpret_cast<uint8_t*>(thdr + 1);
+
+            thdr->sn = sn;
+
+            for (size_t i = 0; i < data_size; ++i) {
+                payload[i] = IoDemoRandom::rand<uint8_t>();
+            }
+
+            dhdr->chk_sum = ucs_crc16(thdr, sizeof(*thdr) + data_size);
+            data_size += sizeof(*dhdr) + sizeof(*thdr);
+        }
+
         send_data_as_chunks(conn, data_size, sn, callback);
     }
-    
+
     void recv_data(UcxConnection* conn, size_t data_size, uint32_t sn,
                    UcxCallback* callback = EmptyCallback::get()) {
+        if (opts().validate) {
+            data_size += sizeof(chk_hdr_t) + sizeof(tr_hdr_t);
+        }
         recv_data_as_chunks(conn, data_size, sn, callback);
     }
 
+    static bool validate(uint32_t sn,      uint32_t peer_sn,
+                         uint16_t chk_sum, uint16_t peer_chk_sum) {
+        bool pass = true;
+
+        if (sn != peer_sn) {
+            pass = false;
```

Comment:
just "return false", no need to have 'pass' var

Reply:
it was done for debug purposes to define data corrupted or transaction miss match or both by log